desktop/bin/build-mac-ci.js

```diff
@@ -13,7 +13,8 @@ const circleTag = process.env.CIRCLE_TAG;
 const isReleaseBuild =
 	process.platform === 'darwin' && !! circleTag && circleTag.startsWith( 'desktop-v' );
 
-const arches = isReleaseBuild ? [ 'x64', 'arm64' ] : [ 'x64' ];
+// Always build both Intel and Apple Silicon versions.
+const arches = [ 'x64', 'arm64' ];
```

Comment:
I've elected to remove the conditional and thus always build both the Intel and Apple Silicon binaries. 

I am thinking that as more of the world moves to Apple Silicon, making sure the arm64 build doesn't break is going to be important.

Reply:
> I've elected to remove the conditional and thus always build both the Intel and Apple Silicon binaries.

I would suggest not doing this, and limiting the build to only one or the other (preferably the arm64 build, as we should be deprecating the Intel build soon anyway).

From the CI run ([link](https://app.circleci.com/pipelines/github/Automattic/wp-calypso/183709/workflows/e0cf0e0a-de82-4043-806e-8b1763f1150c/jobs/1122495)), it's unclear which architecture the end-to-end tests were ran against (as far as I can tell it wasn't both). It's nice that the test script can find either the Apple Silicon or Intel binaries, but you still have to specify which of the two builds to use during a test run (or if you really want to build both, then amend the CI script to execute against both):

```
jest-haste-map: Haste module naming collision: WordPressDesktop
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/release/mac/WordPress.com.app/Contents/Resources/app/package.json
    * <rootDir>/release/mac-arm64/WordPress.com.app/Contents/Resources/app/package.json

 PASS  test/e2e/specs/login.js
  User Can log in
    ✓ Log in (950 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        3.645 s
Ran all test suites.
```

Reply:
> if you really want to build both, then amend the CI script to execute against both

To clarify, the proposed choices are either trim it back to building and testing a single architecture (like the status quo), or build both and run tests for both binaries. 

If so, I can look at getting this out, but with a high-visibility project today and tomorrow I am not sure how much time I have to work on it. 

I'd argue that getting a fix out for users is the priority, as the forum thread is growing.

Reply:
> it sounds like you're in favor of cutting it back to a single arm64 build for non-releases.

Yes - I would lean towards a single architecture for non-release builds:

```
const arches = isReleaseBuild ? [ 'x64', 'arm64' ] : [ 'arm64' ];
```

I would consider the Intel build less important going forward (perhaps it's something we can eventually drop completely).

Reply:
Sounds good, will make the change.